cmd/gendoc/main.go

```diff
@@ -169,32 +170,89 @@ func walkDir(cleanRoot string, md map[string]*DocBlock) error {
 	})
 }
 
-// getPackageName generates full package name from asp.Package
+// findGoMod browses the directory tree starting at the given path
+// and then going up the tree, looking for the first occurrence go.mod file.
+func findGoMod(pkgFilePath string) (goModPath, relPkgPath string, err error) {
+	absPkgFilePath, err := filepath.Abs(pkgFilePath)
+	if err != nil {
+		return "", "", err
+	}
+
+	dirPath := absPkgFilePath
+	for {
+		goModPath = filepath.Join(dirPath, "go.mod")
+		if _, err := os.Stat(goModPath); os.IsNotExist(err) {
+			// This directory does not contain a go.mod file. Go to the parent directory.
+			dirPath = filepath.Dir(dirPath)
```

Comment:
I think we could add a check for reaching root without finding go.mod. Without it, this would cycle indefinitively no ?

Reply:
Good point. I'll have to check it to see what happens when it reaches root. I assumed it would just crash somewhere else at that point due to access violation or something.